src/libratbag-data.c

```diff
@@ -442,6 +450,26 @@ file_data_matches(struct ratbag *ratbag,
 		}
 	}
 
+	data->type = g_key_file_get_string(keyfile, GROUP_DEVICE, "DeviceType", NULL);
+	log_debug(ratbag, "Device Type: '%s'\n", data->type == NULL ? "no type found" : data->type);
+
+	if(data->type == NULL) {
+		// Good idea? Or just add a DeviceType Property to everything
+		log_debug(ratbag, "No DeviceType, assuming Mouse in '%s'\n",  basename(path));
+		data->devicetype = TYPE_MOUSE;
+	} else {
+		if(streq(data->type, "keyboard")) {
+			data->devicetype = TYPE_KEYBOARD;
+		}else if (streq(data->type, "mouse")) {
+			data->devicetype = TYPE_MOUSE;
+		} else {
```

Comment:
you could also just move all devices without a DeviceType here instead of assuming it's a mouse

Reply:
i get a segfault when passing NULL to streq.

Reply:
Ah, that's unlucky. Then nevermind.